src/main/java/org/manifold/compiler/front/ConstantValueVertex.java

```diff
@@ -0,0 +1,56 @@
+package org.manifold.compiler.front;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+
+import org.manifold.compiler.TypeValue;
+import org.manifold.compiler.Value;
+
+public class ConstantValueVertex extends ExpressionVertex {
```

Comment:
Yes, again as I'm sure you mention later, going directly to `ExpressionGraph` should be possible, and will eliminate all of the abstract syntax stuff (including `Expression` and friends). They are suspiciously similar indeed. Opened #48.